### PR TITLE
Replace Bili with Vite

### PR DESCRIPTION
**Pull request by jtoric**
Replaced Bili with Vite.

Also updated all dependencies and moved project to ESM.

Added prettier configuration.

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (droguljic) [inline .prettierrc.js]: This project is set to be an ESM, so this should be changed to `export default` or file renamed to `.cjs`.

File: dist/style.css [hunk 2]
```diff
@@ -0,0 +1 @@
+export default undefined;export default undefined;export default undefined;export default undefined;{}
```
Comment (droguljic) [inline dist/style.css]: What is the purpose of these exports, something seems to be off here.

File: stylelint.config.cjs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  extends: [
+    '@extensionengine/stylelint-config'
+  ],
+  overrides: [
+    {
+      files: ['src/**'],
+      rules: {
+        'selector-pseudo-class-no-unknown': null,
+        'selector-class-pattern': null
```
Comment (droguljic) [inline stylelint.config.cjs]: This can be removed as it is set in `stylelint-config` package.

File: stylelint.config.cjs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  extends: [
+    '@extensionengine/stylelint-config'
+  ],
+  overrides: [
+    {
+      files: ['src/**'],
+      rules: {
+        'selector-pseudo-class-no-unknown': null,
```
Comment (droguljic) [inline stylelint.config.cjs]: What unknown pseudo-class selectors are used that require unsetting this rule?

File: vite.config.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import copy from 'rollup-plugin-copy';
+import postcss from 'rollup-plugin-postcss';
+import { resolve } from 'path';
```
Comment (droguljic) [inline vite.config.js]: 'node:path'

File: vite.config.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import tailorCe from '@extensionengine/rollup-plugin-tailor-ce';
+import vue from '@vitejs/plugin-vue2';
+
+export default {
+  plugins: [
```
Comment (droguljic) [inline vite.config.js:23]: Top level ordering of this config object should be, `build`, `resolve`, `plugins`.

File: vite.config.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import copy from 'rollup-plugin-copy';
+import postcss from 'rollup-plugin-postcss';
+import { resolve } from 'path';
+import tailorCe from '@extensionengine/rollup-plugin-tailor-ce';
+import vue from '@vitejs/plugin-vue2';
+
+export default {
```
Comment (droguljic) [inline vite.config.js:10]: Add type annotation above,
```
/**
 * @type {import('vite').UserConfig}
 */
```

File: package.json [hunk 8]
```diff
@@ -15,44 +13,66 @@
...
+    "postcss-scss": "^4.0.6",
+    "rollup": "^3.26.2",
     "rollup-plugin-copy": "^3.4.0",
-    "rollup-plugin-visualizer": "^4.2.0",
-    "rollup-plugin-vue": "5.1.9",
-    "sass": "^1.53.0",
-    "stylelint": "^13.8.0",
+    "rollup-plugin-postcss": "^4.0.2",
+    "rollup-plugin-visualizer": "^5.9.2",
+    "rollup-plugin-vue": "6.0.0",
```
Comment (droguljic) [inline package.json]: Are these two still used?

File: package.json [hunk 9]
```diff
@@ -15,44 +13,66 @@
...
     "eslint-plugin-n": "^15.2.4",
-    "eslint-plugin-promise": "^6.0.0",
-    "eslint-plugin-vue": "9.2.0",
+    "eslint-plugin-promise": "^6.1.1",
+    "eslint-plugin-require-sort": "^1.3.0",
+    "eslint-plugin-vue": "9.15.1",
+    "eslint-scope": "^7.2.0",
+    "postcss-html": "^1.5.0",
+    "postcss-scss": "^4.0.6",
+    "rollup": "^3.26.2",
```
Comment (droguljic) [inline package.json]: Why is this needed?

File: package.json [hunk 10]
```diff
@@ -15,44 +13,66 @@
...
-    "eslint-config-standard": "^17.0.0",
-    "eslint-plugin-import": "^2.26.0",
+    "del-cli": "^5.0.0",
+    "eslint": "^8.44.0",
+    "eslint-config-prettier": "^8.8.0",
+    "eslint-config-semistandard": "^17.0.0",
+    "eslint-config-standard": "^17.1.0",
+    "eslint-import-resolver-alias": "^1.1.2",
+    "eslint-plugin-import": "^2.27.5",
     "eslint-plugin-n": "^15.2.4",
```
Comment (droguljic) [inline package.json]: Bump to `15.7.0`

File: package.json [hunk 11]
```diff
@@ -15,44 +13,66 @@
...
+    "rollup-plugin-visualizer": "^5.9.2",
+    "rollup-plugin-vue": "6.0.0",
+    "sass": "^1.63.6",
+    "stylelint": "^15.10.1",
+    "vue-eslint-parser": "^9.3.1",
     "vue-runtime-helpers": "^1.1.2",
-    "vue-template-compiler": "^2.7.4"
+    "vue-template-compiler": "^2.7.14"
   },
   "peerDependencies": {
```
Comment (droguljic) [inline package.json:70]: Bump peer dependencies.

File: package.json [hunk 12]
```diff
@@ -15,44 +13,66 @@
...
-    "@extensionengine/eslint-config": "^3.0.1",
-    "@extensionengine/rollup-plugin-tailor-ce": "^0.4.0",
-    "@extensionengine/stylelint-config": "^1.1.1",
-    "@rollup/plugin-alias": "^3.1.9",
+    "@babel/eslint-parser": "^7.22.7",
+    "@babel/eslint-plugin": "^7.22.5",
+    "@extensionengine/eslint-config": "^4.0.1",
+    "@extensionengine/rollup-plugin-tailor-ce": "^0.5.0",
+    "@extensionengine/stylelint-config": "^3.0.0",
+    "@rollup/plugin-alias": "^5.0.0",
```
Comment (droguljic) [inline package.json]: Is this needed?

File: package.json [hunk 13]
```diff
@@ -15,44 +13,66 @@
...
+    "eslint-config-standard": "^17.1.0",
+    "eslint-import-resolver-alias": "^1.1.2",
+    "eslint-plugin-import": "^2.27.5",
     "eslint-plugin-n": "^15.2.4",
-    "eslint-plugin-promise": "^6.0.0",
-    "eslint-plugin-vue": "9.2.0",
+    "eslint-plugin-promise": "^6.1.1",
+    "eslint-plugin-require-sort": "^1.3.0",
+    "eslint-plugin-vue": "9.15.1",
+    "eslint-scope": "^7.2.0",
```
Comment (droguljic) [inline package.json]: Is this used?

File: package.json [hunk 14]
```diff
@@ -15,44 +13,66 @@
...
-    "@extensionengine/stylelint-config": "^1.1.1",
-    "@rollup/plugin-alias": "^3.1.9",
+    "@babel/eslint-parser": "^7.22.7",
+    "@babel/eslint-plugin": "^7.22.5",
+    "@extensionengine/eslint-config": "^4.0.1",
+    "@extensionengine/rollup-plugin-tailor-ce": "^0.5.0",
+    "@extensionengine/stylelint-config": "^3.0.0",
+    "@rollup/plugin-alias": "^5.0.0",
+    "@vitejs/plugin-vue2": "^2.2.0",
     "@vue/compiler-sfc": "^3.2.37",
```
Comment (droguljic) [inline package.json]: Set to `^2.7.14`.

File: package.json [hunk 15]
```diff
@@ -15,44 +13,66 @@
...
+  "umd:main": "./dist/tce-scorm.umd.cjs",
+  "module": "./dist/tce-scorm.js",
+  "exports": {
+    ".": {
+      "import": "./dist/tce-scorm.js",
+      "require": "./dist/tce-scorm.cjs"
+    },
+    "./tce-scorm.css": "./dist/tce-scorm.css"
+  },
+  "type": "module",
```
Comment (droguljic) [inline package.json]: Place above `files`.

File: package.json [hunk 16]
```diff
@@ -3,9 +3,7 @@
   "description": "Use SCORM standard package with Tailor",
   "author": "ExtensionEngine <info@extensionengine.com>",
   "license": "MIT",
-  "version": "0.0.1",
-  "main": "dist/tce-scorm.js",
-  "module": "dist/tce-scorm.esm.js",
+  "version": "0.0.2",
```
Comment (droguljic) [inline package.json]: Revert this change, version will be set after PR is merged, to be inline with other packages.

**Review by droguljic — CHANGES_REQUESTED** [review]

File: .prettierrc.js (hunk 1 above)
Comment (jtoric) [reply .prettierrc.js]: Renamed.

File: dist/style.css (hunk 2 above)
Comment (jtoric) [reply dist/style.css]: Can't be fixed for now.

File: stylelint.config.cjs (hunk 3 above)
Comment (jtoric) [reply stylelint.config.cjs]: Removed.

File: stylelint.config.cjs (hunk 4 above)
Comment (jtoric) [reply stylelint.config.cjs]: Removed.

File: vite.config.js (hunk 5 above)
Comment (jtoric) [reply vite.config.js]: Changed.

File: vite.config.js (hunk 6 above)
Comment (jtoric) [reply vite.config.js:23]: Moved around.

File: vite.config.js (hunk 7 above)
Comment (jtoric) [reply vite.config.js:10]: Added.

File: package.json (hunk 8 above)
Comment (jtoric) [reply package.json]: Removed.

File: package.json (hunk 9 above)
Comment (jtoric) [reply package.json]: Removed.

File: package.json (hunk 10 above)
Comment (jtoric) [reply package.json]: Bumped.

File: package.json (hunk 11 above)
Comment (jtoric) [reply package.json:70]: Bumped.

File: package.json (hunk 12 above)
Comment (jtoric) [reply package.json]: Removed.

File: package.json (hunk 13 above)
Comment (jtoric) [reply package.json]: Removed.

File: package.json (hunk 14 above)
Comment (jtoric) [reply package.json]: Changed.

File: package.json (hunk 15 above)
Comment (jtoric) [reply package.json]: Placed.

File: package.json (hunk 16 above)
Comment (jtoric) [reply package.json]: Reverted.

**Review by droguljic — APPROVED** [review]
